### ISSUE TITLE
xy-plot draws wrong legends when the number of lines > 8

### ISSUE THREAD

**Issue by xfeep** [incanter/incanter#294]
e.g.

```clojure
(def d (doall (for [i (range 9) j (range 2)] 
   {"server" (str "t" i) "conns" (* (inc j) 32) "qps" (* 100 (inc j) (inc i))})) )
(with-data (to-dataset d) 
   (view (xy-plot "conns" "qps" :group-by "server" :legend true :points true)))
```

```clojure
=> (pprint d)
({"server" "t0", "conns" 32, "qps" 100}
 {"server" "t0", "conns" 64, "qps" 200}
 {"server" "t1", "conns" 32, "qps" 200}
 {"server" "t1", "conns" 64, "qps" 400}
 {"server" "t2", "conns" 32, "qps" 300}
 {"server" "t2", "conns" 64, "qps" 600}
 {"server" "t3", "conns" 32, "qps" 400}
 {"server" "t3", "conns" 64, "qps" 800}
 {"server" "t4", "conns" 32, "qps" 500}
 {"server" "t4", "conns" 64, "qps" 1000}
 {"server" "t5", "conns" 32, "qps" 600}
 {"server" "t5", "conns" 64, "qps" 1200}
 {"server" "t6", "conns" 32, "qps" 700}
 {"server" "t6", "conns" 64, "qps" 1400}
 {"server" "t7", "conns" 32, "qps" 800}
 {"server" "t7", "conns" 64, "qps" 1600}
 {"server" "t8", "conns" 32, "qps" 900}
 {"server" "t8", "conns" 64, "qps" 1800})
```

**Comment by xfeep** [comment]
The png result is 
![qps](https://cloud.githubusercontent.com/assets/6120241/6367616/2673e6bc-bd10-11e4-9a1d-0c79e1ee111f.png)

Obviously it is wrong because the line of t8 should be  the highest one.

**Comment by xfeep** [comment]
The version of incanter we used is 

```clojure
[incanter "1.5.6"]
```

**Comment by xfeep** [comment]
If the number of lines <= 8 ,e.g. 

```clojure
(def d2
 (doall 
   (for [i (range 8) j (range 2)] 
     {"server" (str "t" i) "conns" (* (inc j) 32) "qps" (* 100 (inc j) (inc i))})) )
```

```clojure
=> (pprint d2)
({"server" "t0", "conns" 32, "qps" 100}
 {"server" "t0", "conns" 64, "qps" 200}
 {"server" "t1", "conns" 32, "qps" 200}
 {"server" "t1", "conns" 64, "qps" 400}
 {"server" "t2", "conns" 32, "qps" 300}
 {"server" "t2", "conns" 64, "qps" 600}
 {"server" "t3", "conns" 32, "qps" 400}
 {"server" "t3", "conns" 64, "qps" 800}
 {"server" "t4", "conns" 32, "qps" 500}
 {"server" "t4", "conns" 64, "qps" 1000}
 {"server" "t5", "conns" 32, "qps" 600}
 {"server" "t5", "conns" 64, "qps" 1200}
 {"server" "t6", "conns" 32, "qps" 700}
 {"server" "t6", "conns" 64, "qps" 1400}
 {"server" "t7", "conns" 32, "qps" 800}
 {"server" "t7", "conns" 64, "qps" 1600})
```

```clojure
(with-data (to-dataset d2) 
   (view (xy-plot "conns" "qps" :group-by "server" :legend true :points true)))
```

The result of png is right.

![qps2](https://cloud.githubusercontent.com/assets/6120241/6367888/63ab0248-bd12-11e4-8ee6-53a80ad75cf6.png)

**Comment by Gerrrr** [comment]
Thank you for the comprehensive report! The problem is solved in 1.5 and 1.9 branches.

**Comment by nblumoe** [comment]
@Gerrrr @xfeep closing this, as it was fixed by Aleksandr (see above)
